lib/plugins/aws/package/compile/events/cloudWatchLog/index.js

```diff
@@ -15,133 +15,206 @@ class AwsCompileCloudWatchLogEvents {
-                  ] ]
-                },
-                "SourceArn": {
-                  "Fn::Join": [ "", [
-                  "arn:",
-                  { "Ref": "AWS::Partition" },
-                  ":logs:",
-                  { "Ref": "AWS::Region" },
-                  ":",
-                  { "Ref": "AWS::AccountId" },
-                  ":log-group:",
-                  "${LogGroupName}",
-                  ":*"
-                  ] ]
-                }
-              }
-            }
-          `;
+        const newCloudWatchLogRuleObject = {
+          [cloudWatchLogLogicalId]: JSON.parse(cloudWatchLogRuleTemplate),
+        };
 
-            const newCloudWatchLogRuleObject = {
-              [cloudWatchLogLogicalId]: JSON.parse(cloudWatchLogRuleTemplate),
-            };
+        const newPermissionObject = {
+          [lambdaPermissionLogicalId]: JSON.parse(permissionTemplate),
+        };
 
-            const newPermissionObject = {
-              [lambdaPermissionLogicalId]: JSON.parse(permissionTemplate),
-            };
+        _.merge(this.serverless.service.provider.compiledCloudFormationTemplate.Resources,
+          newCloudWatchLogRuleObject, newPermissionObject);
 
-            _.merge(this.serverless.service.provider.compiledCloudFormationTemplate.Resources,
-              newCloudWatchLogRuleObject, newPermissionObject);
-          }
+        // return a new promise that will check the resource limit exceeded against the function
+        return this.checkAndFixLogGroupSubscriptionFilterResourceLimitExceeded({
+          logGroupName: LogGroupName,
+          functionName,
         });
-      }
+      });
+
+      return Promise.all(eventPromises);
+    });
+
+    return Promise.all(promises);
+  }
+
+  /**
+   * @description Cloudwatch imposes a hard limit of 1 subscription filter per log group.
+   * If we change a cloudwatchLog event entry to add a subscription filter to a log group
+   * that already had one before, it will throw an error because CloudFormation firstly
+   * tries to create and replace the new subscription filter (therefore hitting the limit)
+   * before deleting the old one. This precompile process aims to delete existent
+   * subscription filters of functions that a new filter was provided, by checking the
+   * current ARN with the new one that will be generated.
+   * See: https://git.io/fpKCM
+  */
+  checkAndFixLogGroupSubscriptionFilterResourceLimitExceeded({ logGroupName, functionName }) {
+    return new Promise((resolve, reject) => {
+      const region = this.provider.getRegion();
+      const serviceName = this.serverless.service.getServiceName();
+      const stage = this.provider.getStage();
+      const cloudWatchLogs = new this.provider.sdk.CloudWatchLogs({ region });
+
+      cloudWatchLogs
+        .describeSubscriptionFilters({ logGroupName })
+        .promise()
+        .then(({ subscriptionFilters: [subscriptionFilter] }) => {
```

Comment:
What about if a log has multiple subscriptionFilter?

Reply:
So, that's exactly the limitation -- it's not possible to have more than one. I thought about that and didn't understand why this request returns an array and is even paginated if there's a hard limit of a single sub filter.